### PR TITLE
Convert ActionCable tests from CoffeeScript to ES2015 and replace Blade with Karma and Rollup

### PR DESCRIPTION
**Pull request by rmacklin**
### Summary

This is a follow up to #34177, as discussed in https://github.com/rails/rails/pull/34177#issuecomment-429460253. After converting the source to ES2015, we've now converted the tests to ES2015 and replaced Blade with Rollup as the test bundler and Karma as the test runner.

### CODE REVIEW

File: actioncable/test/javascript/src/test_helpers/consumer_test_helper.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import "../../vendor/mock-socket"
```
Comment (javan) [inline actioncable/test/javascript/src/test_helpers/consumer_test_helper.js]: Can we replace vendored [`mock-socket.js`](https://github.com/rails/rails/blob/14e49c2e28e2be4d97d132e99dd92ae9c4ddb4a9/actioncable/test/javascript/vendor/mock-socket.js) file with a `package.json` `devDependency` entry?

File: actioncable/Rakefile [hunk 2]
```diff
@@ -25,12 +25,7 @@ namespace :test do
   end
 
   task :integration do
-    require "blade"
-    if ENV["CI"]
-      Blade.start(interface: :ci)
-    else
-      Blade.start(interface: :runner)
-    end
+    system("yarn install && yarn test") || raise("Failures")
```
Comment (javan) [inline actioncable/Rakefile]: Not loving running `yarn install` from the test command. Let's figure out the Right Way™ and move it to `.travis.yml`. Here-ish?https://github.com/rails/rails/blob/14e49c2e28e2be4d97d132e99dd92ae9c4ddb4a9/.travis.yml#L40-L44 Perhaps @rafaelfranca can offer a little guidance. 😘

File: actioncable/test/javascript/src/test_helpers/consumer_test_helper.js (hunk 1 above)
Comment (rmacklin) [reply actioncable/test/javascript/src/test_helpers/consumer_test_helper.js]: 👍 Replaced in https://github.com/rmacklin/rails/commit/6631c78a3745dc2e6941d77f810df047e81833e4

File: actioncable/Rakefile (hunk 2 above)
Comment (javan) [reply actioncable/Rakefile]: Try removing `yarn install` and adding a line to `.travis.yml` similar to https://github.com/rails/rails/blob/14e49c2e28e2be4d97d132e99dd92ae9c4ddb4a9/.travis.yml#L42

File: actioncable/test/javascript/src/unit/subscription_test.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+module("ActionCable.Subscription", function() {
+  consumerTest("#initialized callback", ({server, consumer, assert, done}) =>
+    consumer.subscriptions.create("chat", {
+      initialized() {
+        assert.ok(true)
+        done()
+      }
+    }
+    )
```
Comment (javan) [inline actioncable/test/javascript/src/unit/subscription_test.js]: These lines should be combined: `})`. Few more like that in this file too.

File: actioncable/test/javascript/src/unit/subscription_test.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
+import consumerTest from "../test_helpers/consumer_test_helper"
+
+const {module} = QUnit
+
+module("ActionCable.Subscription", function() {
+  consumerTest("#initialized callback", ({server, consumer, assert, done}) =>
```
Comment (javan) [inline actioncable/test/javascript/src/unit/subscription_test.js]: Any reason for using `=>` here and `function` on the line above? Can we use `=>` everywhere?

File: actioncable/Rakefile (hunk 2 above)
Comment (rmacklin) [reply actioncable/Rakefile]: Moved `yarn install` to `.travis.yml` in https://github.com/rmacklin/rails/commit/bdc448a959525d10b72be0af8e4114c484cf2228

File: actioncable/test/javascript/src/unit/subscription_test.js (hunk 3 above)
Comment (rmacklin) [reply actioncable/test/javascript/src/unit/subscription_test.js]: Updated in https://github.com/rmacklin/rails/commit/c6eaeb060868b6df9f19fd1153f139b20a69dc37

File: actioncable/karma.conf.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
+const config = {
```
Comment (r-obert) [inline actioncable/karma.conf.js]: just wanted to say the JS is very nice code :)

**Comment by javan** [comment]
Thank you!

**Comment by jeremy** [comment]
🤘🏼

**Pull request merged**